File: #1352.product-of-the-last-k-numbers.cpp
class ProductOfNumbers {
public:
    ProductOfNumbers() {

        nums.push_back(1);
    }

    void add(int num) {

        if (num != 0) {
            nums.push_back(num * nums.back());
            return;
        }
        nums.clear();
        nums.push_back(1);
    }

    int getProduct(int k) {

        if (nums.size() > k) {
            return nums.back() / nums[nums.size() - k - 1];
        }

        return 0;
    }

private:
    vector<int> nums;
};

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers* obj = new ProductOfNumbers();
 * obj->add(num);
 * int param_2 = obj->getProduct(k);
 */

File: #1352.product-of-the-last-k-numbers.js

var ProductOfNumbers = function() {
    this.numbers = [];
    this.length = 0
};

/** 
 * @param {number} num
 * @return {void}
 */
ProductOfNumbers.prototype.add = function(num) {
    this.numbers.push(num);
    this.length++;
};

/** 
 * @param {number} k
 * @return {number}
 */
ProductOfNumbers.prototype.getProduct = function(k) {
    let prod = 1;
    for(let i=this.length-1; i>=0 && k>0; i--){
        prod *= this.numbers[i];
        k--
    }

    return prod;
};

/** 
 * Your ProductOfNumbers object will be instantiated and called as such:
 * var obj = new ProductOfNumbers()
 * obj.add(num)
 * var param_2 = obj.getProduct(k)
 */
